### PR TITLE
add rerank model support for python backend

### PR DESCRIPTION
**Pull request by kaixuanliu**
Add reranker model support for python backend. Test model: [BAAI/bge-reranker-base](https://huggingface.co/BAAI/bge-reranker-base)

### CODE REVIEW

**Comment by kaixuanliu** [comment]
@regisss ， pls help review

**Comment by regisss** [comment]
I see there's already support for it in TEI (see [here](https://github.com/huggingface/text-embeddings-inference?tab=readme-ov-file#using-re-rankers-models)), but it's only for Candle and here you're adding it for Python right?

**Comment by kaixuanliu** [comment]
Yes, python backend does not support "classifier" model.

File: backends/python/server/text_embeddings_server/models/__init__.py [hunk 1]
```diff
@@ -66,10 +67,16 @@ def get_model(model_path: Path, dtype: Optional[str]):
         ):
             return FlashBert(model_path, device, dtype)
         else:
-            return DefaultModel(model_path, device, dtype, trust_remote=TRUST_REMOTE_CODE)
+            if config.architectures[0].endswith("Classification"):
```
Comment (regisss) [inline backends/python/server/text_embeddings_server/models/__init__.py]: I think this check is a bit hacky and not very robust. Maybe a better solution is to check whether `config.model_type` is in https://github.com/huggingface/transformers/blob/165116bc145dcc186fa287e624b28a9ab3a79955/src/transformers/models/auto/modeling_auto.py#L847 ?

File: backends/python/server/text_embeddings_server/models/__init__.py [hunk 2]
```diff
@@ -66,10 +67,16 @@ def get_model(model_path: Path, dtype: Optional[str]):
...
+                return ClassificationModel(model_path, device, dtype)
+            else:
+                return DefaultModel(model_path, device, dtype, trust_remote=TRUST_REMOTE_CODE)
     else:
         try:
-            return DefaultModel(model_path, device, dtype, trust_remote=TRUST_REMOTE_CODE)
+            if config.architectures[0].endswith("Classification"):
```
Comment (regisss) [inline backends/python/server/text_embeddings_server/models/__init__.py]: Same

File: backends/python/server/text_embeddings_server/models/__init__.py (hunk 1 above)
Comment (kaixuanliu) [reply backends/python/server/text_embeddings_server/models/__init__.py]: I was referring to the Rust backend implementations here:[L368](https://github.com/huggingface/text-embeddings-inference/blob/main/router/src/lib.rs#L368). But I think your advice seems better :)

**Review by regisss — APPROVED** [review]
LGTM!

**Pull request merged**